storage/innobase/row/row0ins.cc

```diff
@@ -1391,6 +1393,13 @@ row_ins_foreign_check_on_constraint(
 		}
 	}
 
+	if (table->versioned() && cascade->is_delete != PLAIN_DELETE
+	    && cascade->update->affects_versioned()) {
+		cascade->historical_row =
+			row_build(ROW_COPY_DATA, clust_index, clust_rec, NULL,
+				  table, NULL, NULL, NULL, thr->prebuilt->heap);
```

Comment:
If we allocate this from `row_prebuilt_t::heap`, then the memory consumption will grow until the end of the statement, right?
Do we really need all the `historical_row` so long? What if the statement updated thousands or millions of rows?
And do we need `ROW_COPY_DATA`?